src/analyses/local_safe_pointers.h

```diff
@@ -23,45 +23,17 @@ Author: Diffblue Ltd
 /// possibly-aliasing operations are handled pessimistically.
 class local_safe_pointerst
 {
-  /// Comparator that regards base_type_eq expressions as equal, and otherwise
+  /// Comparator that regards type-equal expressions as equal, and otherwise
   /// uses the natural (operator<) ordering on irept.
-  /// An expression is base_type_eq another one if their types, and types of
-  /// their subexpressions, are identical except that one may use a symbol_typet
-  /// while the other uses that type's expanded (namespacet::follow'd) form.
-  class base_type_comparet
+  class type_comparet
```

Comment:
:question: Does this class even need to exist? Except the ordering is on the exprs, unless the types match - this is somewhat surprising. Out of scope of this PR, but I wonder if this is really what's needed here...

Reply:
It now still is a class, but with a near-trivial inline implementation of the comparison. What approach other than a class would you have thought of?